server/svix-server/src/core/cache/memory.rs

```diff
@@ -0,0 +1,113 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+use tokio::{
+    task,
+    time::{sleep, Duration, Instant},
+};
+
+use axum::async_trait;
+use serde_json;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex, MutexGuard};
+
+use super::{Cache, CacheBehavior, CacheKey, CacheValue, Result};
+
+#[derive(Debug)]
+struct ValueWrapper {
+    value: String,
+    ttl: Duration,
+    timer: Instant,
+}
+
+impl ValueWrapper {
+    fn new(value: String, ttl: Duration) -> ValueWrapper {
+        ValueWrapper {
+            value,
+            ttl,
+            timer: Instant::now(),
+        }
+    }
+}
+
+type State = HashMap<String, ValueWrapper>;
+type SharedState = Arc<Mutex<State>>;
```

Comment:
Use an `RwLock` not a mutex. We read from the cache often, we don't want to be locking everything when just reading.

I also wonder whether there's an atomic way of reading/writing so that a lock needs only to be acquired on writes. Not the end of the world, just a thought.